src/app/Library/CrudPanel/Traits/Fields.php

```diff
@@ -63,6 +59,55 @@ public function makeSureFieldHasNecessaryAttributes($field)
         return $field;
     }
 
+    public function makeSureFieldHasRelationshipAttributes($field)
+    {
+        $field = $this->makeSureFieldHasRelationType($field);
+        $field = $this->makeSureFieldHasModel($field);
+        $field = $this->makeSureFieldHasAttribute($field);
+        $field = $this->makeSureFieldHasMultiple($field);
+        $field = $this->makeSureFieldHasPivot($field);
+        $field = $this->makeSureFieldHasType($field);
+
+        return $field;
+    }
+
+    /**
+     * In case field name is dot notation we want to convert it to a valid HTML array field name for validation purposes.
+     *
+     * @param  array  $field
+     * @return array
+     */
+    public function overwriteFieldNameFromDotNotationToArray($field)
+    {
+        if (! is_array($field['name']) && strpos($field['name'], '.') !== false) {
+            $entity_array = explode('.', $field['name']);
+            $name_string = '';
+
+            foreach ($entity_array as $key => $array_entity) {
+                $name_string .= ($key == 0) ? $array_entity : '['.$array_entity.']';
+            }
+
+            $field['name'] = $name_string;
+        }
+
+        return $field;
+    }
+
+    /**
+     * Run the field name overwrite in multiple fields.
+     *
+     * @param  array  $fields
+     * @return array
+     */
+    public function overwriteFieldNamesFromDotNotationToArray($fields)
+    {
+        foreach ($fields as $key => $field) {
+            $fields[$key] = $this->overwriteFieldNameFromDotNotationToArray($field);
+        }
+
+        return $fields;
+    }
```

Comment:
Any reason these methods have been moved to `Fields` instead of `FieldsProtectedMethods`? If they're `public` but supposed to be `protected`, let's keep them in their old place, that way in v6 we have a clear list of things we need to make `protected` - everything in that file.